server/src/main/java/org/opensearch/index/search/stats/SearchStats.java

```diff
@@ -147,6 +184,29 @@ private Stats(StreamInput in) throws IOException {
                 pitTimeInMillis = in.readVLong();
                 pitCurrent = in.readVLong();
             }
+
+            if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+                this.requestStatsLongHolder = new RequestStatsLongHolder();
+                requestStatsLongHolder.dfsPreQueryMetric = in.readVLong();
+                requestStatsLongHolder.dfsPreQueryCurrent = in.readVLong();
+                requestStatsLongHolder.dfsPreQueryTotal = in.readVLong();
```

Comment:
If `requestStatsLongHolder` was `null` in the instance that was serialized, then this will throw an exception.

Reply:
Good catch. I have changed the logic where we are writing to streamOutput. Now we are always writing to it, so this will never result into exception or case where where values are not available.